CHANGELOG.md

```diff
@@ -1,3 +1,68 @@
+## 2023-01-16 v0.14.0
+
+Release highlights:
+- An important fix for the Waku message archive returning inconsistent responses to history queries.
+- Support for [AutoNAT](https://docs.libp2p.io/concepts/nat/autonat/) and [libp2p Circuit Relay](https://docs.libp2p.io/concepts/nat/circuit-relay/) that allows, among other things, for [NAT hole punching](https://docs.libp2p.io/concepts/nat/hole-punching/).
+- Support for structured logging in JSON format.
+- A fix for an underlying file descriptor leak that affected websocket connections.
+
+### Features
+
+- Support for [AutoNAT](https://docs.libp2p.io/concepts/nat/autonat/)
+- Support for [libp2p Circuit Relay](https://docs.libp2p.io/concepts/nat/circuit-relay/) (server only)
+- New Waku Archive implementation. This allows easy addition of drivers for different technologies to store historical messages.
+- Support for structured logging and specifying log format.
+- Node now keeps track of its external reachability.
+
+### Changes
+
+- Zerokit RLN library now statically linked.
+- Use extended key generation in Zerokit API to comply with [32/RLN](https://rfc.vac.dev/spec/32/).
+- Re-enable root validation in [`17/WAKU-RLN-RELAY`](https://rfc.vac.dev/spec/17/) implementation.
+- [Network monitoring tool](https://github.com/status-im/nwaku/tree/2336522d7f478337237a5a4ec8c5702fb4babc7d/tools#networkmonitor) now supports DNS discovery.
+- Added [dashboard](https://github.com/waku-org/nwaku/blob/3e0e1cb2398297fca761aa74f52d32fa837d556c/metrics/waku-network-monitor-dashboard.json) for network monitoring.
+- Continued refactoring of several protocol implementations to improve maintainability and readability.
+- Removed swap integration from store protocol.
+- Peerstore now consolidated with libp2p peerstore.
+- Peerstore now also tracks peer direction.
+- SIGSEGV signals are now handled and logged properly.
+- Waku v2 no longer imports libraries from Waku v1.
+- Improved build and CI processes:
+  - Added support for an `EXPERIMENTAL` compiler flag.
+  - Simplified project Makefile.
+  - Split Dockerfile into production and experimental stages.
+  - Removed obsolete simulation libraries from build.
+- Improved parallellisation (and therefore processing time) when dialing several peers simultaneously.
```

Comment:
Improved parallellisation (and therefore processing time) when dialing several peers simultaneously. [#1471](https://github.com/waku-org/nwaku/pull/1471)